src/Uno.UI/UI/Xaml/Documents/TextFormatting/FontDetailsCache.skia.cs

```diff
@@ -1,59 +1,205 @@
 ﻿#nullable enable
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
 using HarfBuzzSharp;
 using SkiaSharp;
-using Uno;
 using Uno.Foundation.Logging;
+using Uno.UI;
 using Uno.UI.Xaml;
+using Uno.UI.Xaml.Media;
 using Windows.ApplicationModel;
+using Windows.Storage;
 using Windows.UI.Text;
-using Uno.UI;
 
 namespace Microsoft.UI.Xaml.Documents.TextFormatting;
 
 internal static class FontDetailsCache
 {
-	// TODO: Investigate best value to use here. SKShaper uses a constant 512 scale, Avalonia uses default font scale. Not 100% sure how much difference it
-	// makes here but it affects subpixel rendering accuracy. Performance does not seem to be affected by changing this value.
-	private const int FontScale = 512;
+	private record struct FontCacheEntry(
+		string? Name,
+		float FontSize,
+		FontWeight Weight,
+		FontStretch Stretch,
+		FontStyle Style);
+
+	private static readonly ConcurrentDictionary<string, SKTypeface?> _typefaceCache = new();
+	private static Dictionary<FontCacheEntry, FontDetails> _fontCache = new();
+	private static object _fontCacheGate = new();
+
+	private static JsonSerializerOptions _options = new JsonSerializerOptions()
+	{
+		AllowTrailingCommas = true,
+		PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
+		ReadCommentHandling = JsonCommentHandling.Skip,
+		Converters =
+		{
+			new JsonStringEnumConverter(),
+		},
+	};
+
+	internal static void OnFontLoaded(string font, SKTypeface? typeface)
+	{
+		_typefaceCache[font] = typeface;
+		lock (_fontCacheGate)
+		{
+			foreach (var key in _fontCache.Keys)
+			{
+				if (key.Name == font)
+				{
+					if (_fontCache.TryGetValue(key, out var details))
+					{
+						if (typeface is null)
+						{
+							// font load failed.
+							details.LoadFailed();
+						}
+						else
+						{
+							details.Update(typeface);
+						}
+					}
+				}
+			}
+		}
+	}
+
+	private static string GetFamilyNameFromManifest(Stream jsonStream, FontWeight weight, FontStyle style, FontStretch stretch)
+	{
+		/*
+		 {
+			"fonts": [
+				{
+					"font_style": "normal",
+					"font_weight": 400,
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Regular.ttf"
+				},
+				{
+					"font_style": "italic",
+					"font_weight": "Normal",
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Italic.ttf"
+				},
+				{
+					"font_style": "normal",
+					"font_weight": 700,
+					"font_stretch": "normal",
+					"family_name": "ms-appx:///path/to/ExampleFontFamily-Bold.ttf"
+				}
+			]
+		} 
+		 */
+		var manifest = JsonSerializer.Deserialize<FontManifest>(jsonStream, _options);
```

Comment:
Let's ensure we use generated serializers for this.

Reply:
Will do